### DISCUSSION TITLE
TrustProxies should be configurable

### DISCUSSION THREAD

**Discussion by kolaente** [laravel/framework#38974] [General]
Before https://github.com/laravel/framework/pull/38295, the `TrustProxies` middleware was configurable via an env variable. We dynamically deploy instances of our applications for each PR with docker behind a reverse proxy. Because of this, the IP of the proxy is not known before deploying the application and does not match the IP configured in the `TrustProxies` middleware. This leads to broken URLs (The proxy does TLS termination). To "fix" this we resorted to setting `$proxies = '*'` in the middleware which kind of defeats its whole purpose.

I propose to make the middleware configurable again. However, I guess the removal of the configuration option was intentional, therefore a discussion might be needed.
I could send a PR to add this back in.

**Comment by amenk** [comment]
It's configurable now, but there is no default ENV variable, still
https://github.com/laravel/framework/blob/edde9abaf28b348a658d703965e13530a3dc7673/src/Illuminate/Http/Middleware/TrustProxies.php#L50

PR: https://github.com/laravel/framework/pull/41322
